src-electron/upgrade/upgrade.js

```diff
@@ -40,14 +40,42 @@ async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
   zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data
+  data = JSON.parse(obj)
   let byName = data?.attributeAccessInterfaceAttributes
   let lists = data?.listsUseAttributeAccessInterface
   let forcedExternal = { byName, lists }
   return forcedExternal
 }
 
+/**
+ * This function takes a clusterId and an array of attributes associated with that clusterID from the database
+ * and changes the global attributes (attributes with clusterId = null) to represent storage policy
```

Comment:
Nothing below is restricted to global attributes...

Reply:
Thanks, I just added a check "if(attribute.clusterId == null)". The "non global attributes" should already be computed from the loading process. 

This function should only change global attributes.